### PR TITLE
RFC: Separate secrets from pipelines

### PR DESCRIPTION
**Pull request by elgohr**
Merry christmas :-)

### CODE REVIEW

**Comment by elgohr** [comment]
Hey, thanks for the hint.
This is nearly what I was looking for; I had a little more focus on pipelines, instead of ATC, in my mind.

Especially in big environments you have the need to put more and restrictions in terms of "who's allowed to see which secret?" into the providers configuration. Furthermore centralization was never good for secrets.

At my point of view this could be improved by having the ability to set one provider per team and/or making the providers transparent for the Concourse user by configuring it through the Concourse API. I would prefer the second variant.

File: 001-separate-secrets/proposal.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Summary
+
+> Separate secrets from pipeline definitions
+
+# Motivation
+
+> At the moment secrets and pipelines are mixed together. You can insert a secret via a ((variable)).  
+> In this way you have to set and know every secret on every set-pipeline. There are ways to simplify this, like having it in a file and passing it into the set-pipeline; this makes it much more insecure, as you have to store the file somewhere and/or care for not commiting it to git.  
+
+# Proposal
```
Comment (vito) [inline 001-separate-secrets/proposal.md]: In terms of what this proposes so far I too am a bit confused as it sounds like today's credential management already exists in service of the same goal. There seems to be a lot of overlap between https://github.com/concourse/rfcs/pull/17#issuecomment-450441398 and things being discussed in https://github.com/concourse/rfcs/issues/5 but this RFC doesn't mention any of that.

Maybe this RFC could be steered in a more precise direction based on feedback in #5? Right now it doesn't appear to be proposing any concrete changes and seems to ignore the existing credential manager support in general.

**Comment by elgohr** [comment]
Thank you for mentioning https://github.com/concourse/rfcs/issues/5 . This is going into the direction what I was thinking. I'll close this RFC in favor of discussions in the other RFC

**Comment by vito** [comment]
@elgohr RFCs may be proposed in an early stage so that the community can figure out the details, but they are meant to be centered around changes to propose so that the discussion is more focused and the impact of those changes can be more easily measured. It's difficult to know how to engage if it's too open-ended - reviewers may not know what direction it's trying to go.

RFCs are the step between issues and pull requests:

* 💭 Issue (sharing ideas, developing understanding) ->
* 💡 RFC (a concrete proposal has emerged, let's dig into the details and discuss impact/etc) ->
* ✈️ Pull request (a proposal has been implemented)

I think there are a few things mentioned in #5 that could have proposals made for them (probably as separate RFCs as the domain is quite large). But there hasn't been a *whole* lot of discussion there so that's why I suggested there may be more to do while still in the 'issues' phase, at least for some of the things mentioned (some of it is integration-specific, i.e. Vault).

But if there *is* something that can be proposed concretely that would also be a good start. For example, it feels like we're all in agreement that having per-team credential manager config would be better - the next step for that idea may be an RFC proposing what that may look like. 🙂

**Pull request closed without merging**